messaging-client-databus/src/main/java/com/inmobi/databus/readers/LocalStreamCollectorReader.java

```diff
@@ -281,4 +281,63 @@ protected Date getTimeStampFromCollectorStreamFile(FileStatus file) {
     }
     return null;
   }
+
+  protected Long doRecursiveSizing(Path dir, PathFilter pathFilter) throws Exception {
+    Long pendingSize =0l;
+    FileStatus[] fileStatuses = fsListFileStatus(dir, pathFilter);
+    if (fileStatuses == null || fileStatuses.length == 0) {
+      LOG.debug("No files in directory:" + dir);
+    } else {
+      for (FileStatus file : fileStatuses) {
+        if (file.isDir()) {
+          doRecursiveSizing(file.getPath(), pathFilter);
+        } else {
+          try {
+            Date currentTimeStamp = LocalStreamCollectorReader.
+                    getDateFromStreamFile(streamName, file.getPath().getName());
+            if (stopTime != null && stopTime.before(currentTimeStamp)) {
+              stopListing();
+            } else {
+              pendingSize += file.getLen();
+            }
+          } catch (Exception e) {
+            LOG.error("Exception while getting time from File",e);
+            throw new Exception(e);
+          }
+        }
+      }
+    }
+    return pendingSize;
+  }
+
+  public Long getPendingSize() throws Exception {
+    Long pendingSize = 0l;
+    if (!setBuildTimeStamp(null)) {
+      return 0l;
+    }
+    Calendar current = Calendar.getInstance();
+    Date now = current.getTime();
+    current.setTime(buildTimestamp);
+    // stop the file listing if stop date is beyond current time
+    while (current.getTime().before(now) && !isListingStopped()) {
```

Comment:
why should pending size calculation worry about whether listing is happening by partition readers or stopped? It should just focus on finding the amount of pending data present on HDFS.